City-of-Helsinki/drupal-helfi-etusivu
# UHF-7035: Add elastic endpoint as data-attribute to root element from env

# [UHF-7035](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7035)


## What was done
- Add new config from .env-variable for elastic endpoint
- Embed variable as data-attribute to React root element

## How to install

* Make sure your instance is up and running on correct branch.
* Run `git checkout origin/UHF-7035-endpoint-from-dataset`
* Run `make build`
* Run `make drush-cr`

Add the new env variable: 
* Add this: `ELASTIC_PROXY_URL=https://etusivu-elastic-proxy-dev.agw.arodevtest.hel.fi` to your .env file
* Run `make up` (your app container should restart)
* `make shell` into the container. Run `printenv | grep ELASTIC_PROXY_URL` to verify that the variable is added
* Run `make drush-cr`

## How to test
* Navigate to [news page](https://helfi-etusivu.docker.so/fi/uutiset)
* Check `Fetch/XHR` tab from your network tab in developer tools. `_msearch` -titled requests (Elastic requests) should be going to `https://etusivu-elastic-proxy-dev.agw.arodevtest.hel.fi` 
* You should also see a lot of news results (> 70 pages without any filters) and some weird topics in the topic filter (bad data from news migration, will be fixed by other people)

## Designers review

## Other PRs

* React-app: https://etusivu-elastic-proxy-dev.agw.arodevtest.hel.fi
